src/prefect/agent/ecs/agent.py

```diff
@@ -38,6 +38,14 @@ def merge_run_task_kwargs(opts1: dict, opts2: dict) -> dict:
 
     out = deepcopy(opts1)
 
+    # if opts2 contains keys for capacityProviderStrategy and LaunchType delete launchType
+    if "capacityProviderStrategy" in opts2 and "launchType" in out:
+        del out["launchType"]
```

Comment:
Tried this in prod, found two issues.
1. if we don't specify 'capacityProviderStrategy' in run_task_kwargs and don't specify the default launchType when initiating ecs agent, out["launchType"] = None persists and run_task will throw error as ECS doesn't accept None for launch type.  This prevents us from using the defaultCapacityProvider strategy.

Reply:
I was also concerned about this line. Thanks for giving it a try!

If you have fixes feel free to open another PR that builds on this one. I can't test against these specific cases in ECS right now but I would like to fix up this config merging issues.

Reply:
https://github.com/PrefectHQ/prefect/blob/372fd334e55de12cd37294c33fb32c88d611af94/src/prefect/cli/agent.py#L354-L358

@yin-aven In what case is launch_type getting set as None? How is the ECS Agent being initiated? It should automatically default to FARGATE if it isn't set. 

This line deletes the launchType in the ECSAgent config, if capacityProviderStrategy exists in the ECSRun config and the ECSAgent launchType has already been set. So these lines shouldn't even evaluate if "launchType" is already None.  

https://github.com/PrefectHQ/prefect/blob/84a7e865cf813c0e9bbcaa618bb53daa01c822c5/src/prefect/agent/ecs/agent.py#L46-L47

The reverse applies if capacityProviderStrategy is set in the ECSAgent but launchType is set in the ECSRun.

Reply:
It'd be None if you don't specify --launch-type but set run-task-kwargs-path to a yaml file that contains "capacityProviderStrategy".

But that's not my point.  The real problem is that currently there is no way to use the default capacity provider strategy of a cluster with or without this PR (calling run_task without either launchType or capacityProvideStrategy), which is the most important and reliable way to use ECSAgent IMO.  A wrapper should provide convenience instead of mandating non-required settings.